lib/dotloop/loop.rb

```diff
@@ -2,38 +2,105 @@ module Dotloop
   class Loop
     attr_accessor :client
     BATCH_SIZE = 50
-    # attr_accessor :profile_id
-    # attr_accessor :batch_number
-    # attr_accessor :batch_size
-    # attr_accessor :status_ids
-    # attr_accessor :compliance_status_ids
-    # attr_accessor :tag_ids
-    # attr_accessor :sort_by
-    # attr_accessor :search_query
-    # attr_accessor :tag_names
-    # attr_accessor :created_by_me
+    MAX_LOOPS = 500
 
     def initialize(client:)
       @client = client
     end
 
-    def all(profile_id:)
-      batches = []
-      batch_number = 1
-      loop do
-        batch = retrieve_batch(profile_id: profile_id, batch_size: BATCH_SIZE, batch_number: batch_number)
-        batches += batch
-        batch_number += 1
-        break if batch.size < BATCH_SIZE
+    # rubocop:disable Metrics/ParameterLists
+    def all(
+      profile_id:, batch_number: nil, batch_size: nil, status_ids: [],
+      compliance_status_ids: [], tag_ids: [], sort_by: nil, search_query: nil,
+      tag_names: nil, created_by_me: nil
+    )
+      options = keyword_to_hash(__method__, binding)
+      options[:batch_size] = BATCH_SIZE
+      loops = []
+      (1..MAX_LOOPS).each do |i|
+        options[:batch_number] = i
+        current_loop = batch(options)
+        loops += current_loop
+        break if current_loop.size < BATCH_SIZE
       end
-      batches
+      loops
+    end
+
+    def batch(
+      profile_id:, batch_number: nil, batch_size: nil, status_ids: [],
+      compliance_status_ids: [], tag_ids: [], sort_by: nil, search_query: nil,
+      tag_names: nil, created_by_me: nil
```

Comment:
Hmm....not exactly what I meant 😅 

How about:

```
def batch(*args)
    @client.get("/profile/#{profile_id.to_i}/loop", query_params(args)).map do |attrs|
        Dotloop::Models::Loop.new(attrs)
    end
end
```

The options will be "validated" because they'll be merged in the ```query_params``` call.